### PR TITLE
Allow Dockerfile to be written inline

### LINKED ISSUE #32: Provide steps in YAML instead of Dockerfile
I'd like to share a common pipeline between several repositories, where the actual repositories' YAML is `aws s3 cp ... - | bash | buildkite-pipeline upload` and the script stored in S3 does the work. And to speed up the builds, I'd like to utilise the ECR caching in the common pipeline. However, the repositories won't know any of this and will not have a Dockerfile to enable the plugin to work.

I'd like to be able to specify something like this _instead_ of a Dockerfile:

```yaml
steps:
  - command: npm test
    plugins:
      - seek-oss/docker-ecr-cache#v1.9.0:
          cache-on:
            - package-lock.json
          base-image: node:10-alpine
          steps:
            - npm install
            - curl large_file ...
            - ...
```

Another option that would solve this problem is specifying a Dockerfile location in S3.

### PR DESCRIPTION
**Pull request by koterpillar**
Fixes #32 

Adds new `dockerfile-inline` configuration option to write Dockerfile inside the plugin options.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,6 +89,23 @@ steps:
...
+```yaml
+steps:
+  - command: echo wow
+    plugins:
+      - seek-oss/docker-ecr-cache#v1.9.0:
+          dockerfile-inline: |
+            FROM node:16-alpine
+            WORKDIR /workdir
+            COPY package.json package-lock.json /workdir
+            RUN npm install
```
Comment (72636c) [inline README.md:103]: Are there any catches with Buildkite variable interpolation that may be worth documenting?

File: hooks/pre-command [hunk 2]
```diff
@@ -9,12 +9,21 @@ source "$(dirname "${BASH_SOURCE[0]}")/lib/${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_R
 login
 configure_registry_for_image_if_necessary
 image="$(get_registry_url)"
-docker_file="${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_DOCKERFILE:-"Dockerfile"}"
+if [ -n "${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_DOCKERFILE_INLINE-}" ]
```
Comment (72636c) [inline hooks/pre-command]: NBD but we seem to use the "unset or empty" syntax elsewhere:

```suggestion
if [ -n "${BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_DOCKERFILE_INLINE:-}" ]
```

**Review by 72636c — APPROVED** [review]
Awesome, thanks for this!

**Comment by koterpillar** [comment]
I still have a few things to work out in how this integrates with BuildKit (#36)...

File: README.md (hunk 1 above)
Comment (koterpillar) [reply README.md:103]: I couldn't find anything, seems that `|` is very forgiving.

**Comment by koterpillar** [comment]
This and #36 both ready and tested internally.

**Review by 72636c — APPROVED** [review]

**Pull request merged**